pkg/proxy/handler_test.go

```diff
@@ -1,69 +1,157 @@
 package proxy
 
 import (
+	"context"
+	"fmt"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"strings"
 	"testing"
 
-	"github.com/go-logr/logr"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/testutil"
 	io_prometheus_client "github.com/prometheus/client_model/go"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"github.com/substratusai/lingo/pkg/deployments"
-	"k8s.io/apimachinery/pkg/runtime"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
-
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/cache"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/config"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-func TestMetrics(t *testing.T) {
+func TestHandler(t *testing.T) {
+	const (
+		model1 = "model1"
+		model2 = "model2"
+
+		maxRetries = 3
+	)
+	models := map[string]string{
+		model1: "deploy1",
+		model2: "deploy2",
+	}
+
 	specs := map[string]struct {
-		request   *http.Request
-		expCode   int
-		expLabels map[string]string
+		reqMethod string
+		reqPath   string
```

Comment:
personal preference: reqMethod + reqPath are the same in all tests. It would be more readable to move them to the call and reduce the number of options in the test.

Reply:
Agreed